src/DotNetWorker.Grpc/GrpcWorker.cs

```diff
@@ -223,7 +223,10 @@ private async Task ProcessRequestCoreAsync(StreamingMessage request)
             }
             finally
             {
-                (context as IDisposable)?.Dispose();
+                if (context is IAsyncDisposable asyncContext)
```

Comment:
As discussed, we need to maintain the behavior where we dispose of `IDisposable` only contexts, as this change would prevent that from happening.

Reply:
Brought back the code which handles the case of FunctionContext being just `IDisposable` type.